### ISSUE TITLE
[rest/sse] Not able to subscribe to sitemap events

### ISSUE THREAD

**Issue by kaikreuzer** [openhab/openhab-core#1585]
When UIs try to subscribe to sitemap events through `/rest/sitemaps/events/subscribe`, I am seeing HTTP 503 being returned with body
```
{
  "error": {
    "message": "Max number of subscriptions is reached.",
    "http-code": 503
  }
}
```
It seems that there is some regression as I am only using 1 or two clients at a time.

**Comment by Donokami** [comment]
I've got the same issue, quite annoying

**Comment by wborn** [comment]
Did you already investigate this issue by monitoring the subscriber count and cleanup by enabling debug logging on `org.openhab.core.io.rest.sitemap`? It will only return this error when the limit of 50 subscriptions is reached. So it should work for some time after restarting openHAB.

**Comment by kaikreuzer** [comment]
No, I didn't yet find the time to analyse the issue. Will try to do it soon®.

**Comment by wborn** [comment]
I did have some time now to look into this and after enabling the debug logging on `org.openhab.core.io.rest.sitemap` it looks like the openHAB Android App is creating a new SSE sitemap subscription whenever you navigate to different sitemap pages. Also if you switch to another App and then back to the openHAB Android App it creates new sitemap subscriptions. So I can imagine those 50 subscriptions are exhausted pretty quickly.

Perhaps the @openhab/android-maintainers can look into this? Basic UI reuses the existing subscriptionId so it doesn't cause such issues.

**Comment by kaikreuzer** [comment]
Thanks @wborn! I have finally found the time to test it myself and I can absolutely confirm your analysis. When restarting my instance, it indeed first works and with every access through the Android app, I see the number of subscriptions to increase - even when navigating to new pages, which must definitely be considered as a bug.

I'd hence suggest to move this issue over to the Android repo.

Ftr, this is what I see in the log when changing a page in the Android app:
```
21:29:42.702 [DEBUG] [st.sitemap.SitemapSubscriptionService] - Created new subscription with id f8206432-4a4f-4434-afa3-c08efb3d02a0 (8 active subscriptions for a max of 50)
21:29:42.703 [DEBUG] [rest.sitemap.internal.SitemapResource] - Client from IP 192.168.0.164 requested new subscription => got id f8206432-4a4f-4434-afa3-c08efb3d02a0.
21:29:42.713 [DEBUG] [rest.sitemap.internal.SitemapResource] - Received HTTP GET request from IP 192.168.0.164 at 'sitemaps/default/0203'
21:29:42.736 [DEBUG] [st.sitemap.SitemapSubscriptionService] - Subscription f8206432-4a4f-4434-afa3-c08efb3d02a0 changed to page 0203 of sitemap default (8 active subscriptions}
21:29:42.748 [DEBUG] [rest.sitemap.internal.SitemapResource] - Client from IP 192.168.0.164 requested sitemap event stream for subscription f8206432-4a4f-4434-afa3-c08efb3d02a0.
```
While this is what happens when using the Basic UI:
```
21:33:22.152 [DEBUG] [st.sitemap.SitemapSubscriptionService] - Subscription 3d34ae5c-0918-4d7a-ab82-4a79f97a6ed8 changed to page 0102 of sitemap default (8 active subscriptions}
```
